Assets/Scripts/CombatMovement.cs

```diff
@@ -116,17 +138,23 @@ protected void Move()
                 transform.position = target;
                 if (path.Count == 1)
                 {
+                    currentTile.isBlocked = false;
                     currentTile.isCurrent = false;
                     currentTile = path.Peek();
                     currentTile.isCurrent = true;
+                    currentTile.isBlocked = true;
+                }
+                else
+                {
+                    actionPoints -= 1;
```

Comment:
Great question.

Basically, the way the search is set up right now, both the origin tile and the destination tile are considered part of the movement path, so the overall number of tiles covered by the movement path is the number of tiles moved plus one.

So, by discounting one of the tiles from the cost (either the origin or the destination, in this case I chose the destination), it ensures that the cost in action points is equal to the number of tiles moved.

Do you think it would be more clear to discount the origin tile instead of the destination tile? And possibly also add a comment explaining the reasoning?

Reply:
I should have thought of that. Yeah, I think a comment would be helpful.